File: exercism/elixir/anagram/anagram.exs
defmodule Anagram do
  @doc """
  Returns all candidates that are anagrams of, but not equal to, 'base'.
  """
  @spec match(String.t, [String.t]) :: [String.t]
  def match(base, candidates) do
    candidates |> Enum.filter(&
      String.downcase(base) != String.downcase(&1) and
      canonical(base) == canonical(&1))
  end

  defp canonical(str) do
    str |> String.downcase
        |> String.to_char_list
        |> Enum.sort
  end
end

File: exercism/ruby/anagram/anagram.rb
module BookKeeping
  VERSION = 2
end

class String
  def sort
    chars.sort.join
  end
  def anagram? s
    self != s && sort == s.sort
  end
end

class Anagram
  def initialize(str)
    @str = str.downcase
  end
  def match(ss)
    ss.select{|s| @str.anagram?(s.downcase)}
  end
end
